### PR TITLE
Delete article

### PR DESCRIPTION
**Pull request by AlenGano**
https://github.com/Northeastern-Science-Magazine/nusci-backend/pull/283

### CODE REVIEW

File: app/models/dbModels/article.js [hunk 1]
```diff
@@ -53,6 +53,19 @@ const ArticleSchema = new Schema(
...
+
+ArticleSchema.pre('deleteOne', { document: true, query: false }, async function() {
+  try {
+    // changed name to IssueMapSchema 
+    await mongoose.model('IssueMap').updateMany(
+      { articles: this._id },
+      { $pull: { articles: this._id } }
+    );
+  } catch (e) {
+    console.error('Error in pre-deleteOne hook:', e);
```
Comment (Arushi55) [inline app/models/dbModels/article.js:65]: throw an actual error?

File: app/models/dbModels/article.js [hunk 2]
```diff
@@ -53,6 +53,19 @@ const ArticleSchema = new Schema(
     collection: "articles",
   }
 );
+
+ArticleSchema.pre('deleteOne', { document: true, query: false }, async function() {
+  try {
+    // changed name to IssueMapSchema
```
Comment (Arushi55) [inline app/models/dbModels/article.js:59]: comments about changes from previous iterations seem unnecessary

File: app/databaseAccessors/articleAccessor.js [hunk 3]
```diff
@@ -328,4 +329,37 @@ export default class ArticlesAccessor {
...
+
+    const article = await Article.findOne( {slug: slug} );
+    if (article) {
+
+      try {
+      console.log("we found the article");
+
+      //console.log(article);
+
+      const resp = await Article.deleteOne({ _id: article._id });
```
Comment (Arushi55) [inline app/databaseAccessors/articleAccessor.js:348]: If this calls the delete hook for an issue map, when is this article actually deleted from the articles list itself?

**Review by Arushi55 — CHANGES_REQUESTED** [review]
Some things to think about, especially to test whether or not the article itself is every actually deleted.
The hook is really neat though!

**Review by ethanszeto — CHANGES_REQUESTED** [review]
Please rebase to main, the git diff is +4200 and you have 105 files changed. After that we can review

**Comment by ethanszeto** [comment]
https://github.com/Northeastern-Science-Magazine/nusci-backend/pull/283

**Pull request closed without merging**